File: src/commands/MaxByFullNameCommand.java
package commands;

import commands.core.ServerSideCommand;
import lib.ExecutionStatus;
import lib.functions.CallbackFunction;
import organization.Organization;
import organization.OrganizationManager;

import java.util.Collections;
import java.util.Comparator;

public class MaxByFullNameCommand extends ServerSideCommand {

    public MaxByFullNameCommand(CallbackFunction callbackFunction, OrganizationManager organizationManager) {
        super(callbackFunction, organizationManager);
    }

    @Override
    protected void executeImplementation(String[] args, CallbackFunction callback) {
        Organization maxOrganization = Collections.max(organizationManager.getOrganizations(),
                Comparator.comparing(Organization::getFullName));

        if (maxOrganization != null) {
            callback.invoke(ExecutionStatus.SUCCESS, null, maxOrganization);
        } else {
            callback.invoke(ExecutionStatus.FAILURE, null, (Organization) null);
        }
    }
}

File: src/commands/client_side/MaxByFullNameCommand.kt
package commands.client_side;

import OrganizationDatabase.Organization;
import OrganizationDatabase.OrganizationManagerInterface;
import commands.client_side.core.ServerSideCommand;
import exceptions.OrganizationNotFoundException;
import lib.ExecutionStatus;
import commands.client_side.core.ClientCallbackFunction;

public class MaxByFullNameCommand extends ServerSideCommand {

    public MaxByFullNameCommand(ClientCallbackFunction clientCallbackFunction, OrganizationManagerInterface organizationDatabase) {
        super(clientCallbackFunction, organizationDatabase);
    }

    @Override
    protected void executeImplementation(String[] args, ClientCallbackFunction callback) throws OrganizationNotFoundException {
        Organization maxOrganization = organizationDatabase.maxByFullName();

        if (maxOrganization != null) {
            callback.invoke(ExecutionStatus.SUCCESS, null, maxOrganization);
        } else {
            callback.invoke(ExecutionStatus.FAILURE, null);
        }
    }
}
